content/guides/quickstart/voting-guide.md

```diff
@@ -0,0 +1,1230 @@
+::
+++  on-arvo
+  |=  [=wire =sign-arvo]
+  ^-  (quip card _this)
+  ?.  ?=([%bind ~] wire)
+    (on-arvo:def [wire sign-arvo])
+  ?.  ?=([%eyre %bound *] sign-arvo)
+    (on-arvo:def [wire sign-arvo])
+  ~?  !accepted.sign-arvo
+    %eyre-rejected-tally-binding
+  `this
+::
+++  on-leave  on-leave:def
+++  on-peek   on-peek:def
+++  on-fail   on-fail:def
+--
+|_  bol=bowl:gall
+++  make-index
+  |=  rid=@ta
+  ^-  (list card)
+  %+  make-200
+    rid
+  %-  as-octs:mimes:html
+  %-  crip
+  %-  en-xml:html
+  (index bol by-group voted withdrawn)
+::
+++  make-200
+  |=  [rid=@ta dat=octs]
+  ^-  (list card)
+  %^    give-http
+      rid
+    :-  200
+    :~  ['Content-Type' 'text/html']
+        ['Content-Length' (crip ((d-co:co 1) p.dat))]
+    ==
+  [~ dat]
+::
+++  give-http
+  |=  [rid=@ta hed=response-header:http dat=(unit octs)]
+  ^-  (list card)
+  :~  [%give %fact ~[/http-response/[rid]] %http-response-header !>(hed)]
+      [%give %fact ~[/http-response/[rid]] %http-response-data !>(dat)]
+      [%give %kick ~[/http-response/[rid]] ~]
+  ==
+::
+++  make-ring-members
+  |=  =gid
+  ^-  (set [=ship =life])
+  =/  invited=(list @p)  ~(tap in (get-members gid))
+  =|  members=(set [=ship =life])
+  |-
+  ?~  invited
+    members
+  =/  lyfe
+    .^  (unit @ud)
+      %j
+      (scot %p our.bol)
+      %lyfe
+      (scot %da now.bol)
+      /(scot %p i.invited)
+    ==
+  ?~  lyfe
+    $(invited t.invited)
+  %=  $
+    invited  t.invited
+    members  (~(put in members) [i.invited u.lyfe])
+  ==
+::
+++  get-members
+  |=  =gid
+  ^-  (set ship)
+  %-  ~(gas in *(set ship))
+  %+  skim
+    %~  tap  in
+    =<  members
+    %-  fall
+    :_  *group:g
+    .^  (unit group:g)
+      %gx
+      (scot %p our.bol)
+      %group-store
+      (scot %da now.bol)
+      /groups/ship/(scot %p p.gid)/[q.gid]/noun
+    ==
+  |=  =ship
+  ?|  =(our.bol ship)
+      ?=(?(%czar %king %duke) (clan:title ship))
+  ==
+--
+```
+
+### Marks
+
+Marks are Urbit's version of filetypes/MIME types. We need to define a mark for
+the `action`s we'll send or receive, and the `update`s we'll send to
+subscribers or receive for subscriptions. These will be very simple since we
+don't need to do any conversions to things like JSON.
+
+Mark files are stored in the `/mar` directory of a desk. Save the
+`%tally-action` mark in `tally/mar/tally/action.hoon`, and the `%tally-update`
+mark in `tally/mar/tally/update.hoon`.
+
+#### `%tally-action`
+
+```hoon {% copy=true %}
+/-  *tally
+|_  act=action
+++  grow
+  |%
+  ++  noun  act
+  --
+++  grab
+  |%
+  ++  noun  action
+  --
+++  grad  %noun
+--
+```
+
+#### `%tally-update`
+
+```hoon {% copy=true %}
+/-  *tally
+|_  upd=update
+++  grow
+  |%
+  ++  noun  upd
+  --
+++  grab
+  |%
+  ++  noun  update
+  --
+++  grad  %noun
+--
+```
+
+### Front-end
+
+We could have put the front-end code directly in our Gall agent, but it tends to
+be quite large so it's convenient to have it in a separate file and just import
+it.
```

Comment:
“Most of this file consists of Sail code, which is the internal HTML representation, similar to other server-side renderings like Clojure's Hiccup.”